vrouter.go

```diff
@@ -397,6 +480,78 @@ func (self *Vrouter) UpdateLocalEndpoint(endpoint *OfnetEndpoint, epInfo Endpoin
 	return nil
 }
 
+// AddHostPort sets up host access.
+func (self *Vrouter) AddHostPort(hp HostPortInfo) error {
+	if hp.Kind == "NAT" && self.hostNATInfo.PortNo != 0 {
+		log.Errorf("Host NAT port exists: %+v", self.hostNATInfo)
+		return fmt.Errorf("Host NAT port exists")
+	}
+
+	ipDa, daMask, err := ParseIPAddrMaskString(hp.IpAddr)
+	if err != nil {
+		log.Errorf("Bad host route - %v", err)
+		return err
+	}
+
+	netMask := pflag.ParseIPv4Mask(daMask.String())
+	maskedIP := ipDa.Mask(netMask)
+
+	// Save the info
+	self.hostNATInfo = hp
```

Comment:
I thought I saw portNo in hostNATInfo getting manipulated here. Double checked again and saw that it was only during the remove. If we are not modifying anything here and not maintaining its reference in netplugin, a pointer is not needed. Please ignore my nil check comments as well in that case.